core/primitives/src/types.rs

```diff
@@ -905,6 +905,8 @@ pub enum EpochReference {
     Latest,
 }
 
+pub type MaybeEpochReference = Option<EpochReference>;
```

Comment:
I'm not in favor of defining types for `Option`s. The resulting code is almost the same length, but introduces an additional concept.